src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,176 @@
+
+abstract contract VeTokenManager is Core {
+  event BuyBoost(address delegator, address receiver, uint256 amount, uint256 duration);
+  event DelegateUpdate(address indexed oldDelegate, address indexed newDelegate);
+  event Lock(uint256 cummulativeTokensLocked, uint256 lockHorizon);
+  event Unlock(uint256 tokensUnlocked);
+
+  DelegateRegistry public constant DELEGATE_REGISTRY =
+    DelegateRegistry(0x469788fE6E9E9681C6ebF3bF78e7Fd26Fc015446);
+
+  address public constant BAL = 0xba100000625a3754423978a60c9317c58a424e3D;
+  address public constant B_80BAL_20WETH = 0x5c6Ee304399DBdB9C8Ef030aB642B10820DB8F56;
+  address public constant VE_BAL = 0xC128a9954e6c874eA3d62ce62B468bA073093F25;
+  address public constant WARDEN_VE_BAL = 0x42227bc7D65511a357c43993883c7cef53B25de9;
+
+  /// @notice the snapshot delegate for the deposit
+  address public delegate;
+  /// @notice the keccak encoded spaceId of the snapshot space
+  bytes32 public spaceId;
+  /// @notice The lock duration of veToken
+  uint256 lockDuration;
+
+  /// @notice The Buyer will then need to call the user_checkpoint() method on all Curve V4 & Factory Gauges
+  /// currently farming, to apply the newly received Boost.
+  /// @notice duration in weeks (ie: 1 for 1 Week)
+  function buyBoost(
+    address delegator,
+    address receiver,
+    uint256 amount,
+    uint256 duration
+  ) external onlyOwnerOrGuardian {
+    uint256 maxFee = IWardenBoost(WARDEN_VE_BAL).estimateFees(delegator, amount, duration);
+    IERC20(BAL).approve(WARDEN_VE_BAL, maxFee);
+    IWardenBoost(WARDEN_VE_BAL).buyDelegationBoost(delegator, receiver, amount, duration, maxFee);
+
+    emit BuyBoost(delegator, receiver, amount, duration);
+  }
+
+  /// @notice Sells boost in order to accrue rewards
+  /// @dev Registers a new user, creates a BoostOffer with the given parameters
+  /// @param pricePerVote Price of 1 vote per second (in wei)
+  /// @param maxDuration Maximum duration (in weeks) that a Boost can last when taken from this Offer
+  /// @param expiryTime Timestamp when this Offer is not longer valid
+  /// @param minPerc Minimum percent of users voting token balance to buy for a Boost (in BPS)
+  /// @param maxPerc Maximum percent of users total voting token balance available to delegate (in BPS)
+  /// @param useAdvicePrice True to use the advice Price instead of the given pricePerVote
+  function sellBoost(
+    uint256 pricePerVote,
+    uint64 maxDuration,
+    uint64 expiryTime,
+    uint16 minPerc,
+    uint16 maxPerc,
+    bool useAdvicePrice
+  ) external onlyOwnerOrGuardian {
+    IERC20(IWardenBoost(WARDEN_VE_BAL).delegationBoost()).approve(
+      WARDEN_VE_BAL,
+      type(uint256).max // Per Boost docs, approve uint256 max
+    );
+    IWardenBoost(WARDEN_VE_BAL).register(
+      pricePerVote,
+      maxDuration,
+      expiryTime,
+      minPerc,
+      maxPerc,
+      useAdvicePrice
+    );
+  }
+
+  /// @notice Update an existing boost offer
+  function updateBoostOffer(
```

Comment:
Missing NatSpec, (see previous comments)

Reply:
Same for some other functions

Reply:
Done.